connector/openshift/openshift_test.go

```diff
@@ -184,6 +184,57 @@ func TestCallbackIdentity(t *testing.T) {
 	expectEquals(t, identity.Groups[0], "users")
 }
 
+func TestQueryOAuth2Endpoints(t *testing.T) {
+	s := newTestServer(map[string]interface{}{
+		"/oauth/authorize": nil,
+		"/oauth/token": nil,
+	})
+	defer s.Close()
+
+	hostURL, err := url.Parse(s.URL)
+	expectNil(t, err)
+
+	_, err = http.NewRequest("GET", hostURL.String(), nil)
+	expectNil(t, err)
+
+	h, err := newHTTPClient(true, "", nil)
+	expectNil(t, err)
+
+	oc := openshiftConnector{apiURL: s.URL, httpClient: h, oauth2Config: &oauth2.Config{
+		Endpoint: oauth2.Endpoint{
+			AuthURL: s.URL + "/oauth/authorize", TokenURL: s.URL + "/oauth/token",
+		},
+	}}
+	err = oc.queryOAuth2Endpoints()
+	expectNil(t, err)
+}
+
+func TestQueryOauth2EndpointsFailure(t *testing.T) {
+	s := newTestServer(map[string]interface{}{
+		"/oauth/authorize": nil,
+		"/oauth/token": nil,
+	})
+	defer s.Close()
+	hostURL, err := url.Parse(s.URL)
+	expectNil(t, err)
+
+	_, err = http.NewRequest("GET", hostURL.String(), nil)
+	expectNil(t, err)
+
+	s.Close()
```

Comment:
This part has me confused. Why? I see it's in line with the other tests, but it looks like literally nothing related to oauth2 endpoints is being tested here?

Reply:
I honestly just did it to be in line with the other tests, where always once the test server was started, a request created and verified that it _can be_ created. I don't know whether this makes much sense to keep it that way, I would keep it for now, ask during the upstream review and potentially send an additional PR cleaning up _all tests_ if we deem it unnecessary